main-presentation/src/main/java/com/teamhy2/feature/home/FocusModeScreen.kt

```diff
@@ -0,0 +1,251 @@
+                timerViewModel.stopTimer()
+                localNavController.popUpToHome()
+                tracker.trackEvent("StudyEndButton")
+            },
+            onDismiss = {
+                isStudyRoomEndDialog = false
+            },
+        )
+    }
+
+    when (timerState) {
+        TimerUiState.Idle -> HY2LoadingScreen()
+        is TimerUiState.Running -> {
+            FocusModeScreen(
+                timerUiState = timerState as TimerUiState.Running,
+                onStudyRoomEndClick = {
+                    isStudyRoomEndDialog = true
+                },
+                onStudyRoomExtendClick = {
+                    isStudyRoomExtendDialog = true
+                },
+                onCloseClick = { localNavController.popBackStack() },
+                modifier = modifier,
+            )
+        }
+    }
+}
+
+@Composable
+fun FocusModeScreen(
+    timerUiState: TimerUiState.Running,
+    onStudyRoomEndClick: () -> Unit,
+    onStudyRoomExtendClick: () -> Unit,
+    onCloseClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Column(modifier = modifier) {
+        Box(
+            modifier =
+                Modifier
+                    .fillMaxWidth()
+                    .height(52.dp),
+            contentAlignment = Alignment.CenterEnd,
+        ) {
+            IconButton(onClick = onCloseClick) {
+                Icon(
+                    painter = painterResource(com.teamhy2.designsystem.R.drawable.ic_close),
+                    contentDescription = "닫기",
+                    tint = Gray100,
+                )
+            }
+        }
+        Column(
+            modifier =
+                Modifier
+                    .padding(horizontal = 32.dp)
+                    .offset(y = (-12).dp),
+        ) {
+            HY2Timer(
+                durationAsSecond = timerUiState.duration.seconds,
+                leftTime = timerUiState.leftTime.value,
+                startTime = timerUiState.startTime.formattedTime,
+                startTimeMeridiem = timerUiState.startTime.meridiem.label,
+                endTime = timerUiState.endTime.formattedTime,
+                endTimeMeridiem = timerUiState.endTime.meridiem.label,
+            )
+            Spacer(modifier = Modifier.height(28.dp))
+            Row {
+                HY2Button(
+                    text = "열람실 이용 종료",
+                    onClick = onStudyRoomEndClick,
+                    textColor = Gray100,
+                    backgroundColor = Gray600,
+                    modifier = Modifier.weight(1f),
+                )
+                Spacer(modifier = Modifier.width(12.dp))
+                HY2Button(
+                    text = "열람실 이용 연장",
+                    onClick = onStudyRoomExtendClick,
+                    modifier =
+                        Modifier
+                            .weight(1f)
+                            .alpha(if (timerUiState.leftTime.value <= "00:30:00") 1f else 0f),
```

Comment:
```suggestion
                            .alpha(1f.takeIf { timerUiState.leftTime.value <= "00:30:00" } ?: 0f),
```
takeIf라는 확장함수가 있는데 람다 안에 조건이 맞으면 나를 쓰고 아니면 null을 반환해라 라는 함수인데 함수형 프로그래밍에서 자주 사용한다고 시니어 개발자가 말해주었는데 어떠신가요? if/else가 저도 아직 눈에 익긴합니다

Reply:
오 적극적으로 찬성합니다. 좀더 코틀린스럽다 느껴지네요.